src/documents/tasks.py

```diff
@@ -319,3 +324,92 @@ def empty_trash(doc_ids=None):
             cleanup_document_deletion,
             sender=Document,
         )
+
+
+@shared_task
+def check_scheduled_workflows():
+    scheduled_workflows: list[Workflow] = Workflow.objects.filter(
+        triggers__type=WorkflowTrigger.WorkflowTriggerType.SCHEDULED,
+        enabled=True,
+    ).prefetch_related("triggers")
+    if scheduled_workflows.count() > 0:
+        logger.debug(f"Checking {len(scheduled_workflows)} scheduled workflows")
+        for workflow in scheduled_workflows:
+            schedule_triggers = workflow.triggers.filter(
+                type=WorkflowTrigger.WorkflowTriggerType.SCHEDULED,
+            )
+            trigger: WorkflowTrigger
+            for trigger in schedule_triggers:
+                documents = Document.objects.none()
+                offset_td = timedelta(days=trigger.schedule_offset_days)
+                logger.debug(
+                    f"Checking trigger {trigger} with offset {offset_td} against field: {trigger.schedule_date_field}",
+                )
+                if (
```

Comment:
Now that we're on just 3.10, this  could use `match`.   I think it would look cleaner

Reply:
Oh yea cool, I didn't really remember this one, so nice to have now!